packages/composite-checkout/src/components/checkout-error-boundary.tsx

```diff
@@ -16,19 +16,19 @@ export default class CheckoutErrorBoundary extends React.Component< CheckoutErro
 
 	public state = { hasError: false };
 
-	static getDerivedStateFromError() {
+	static getDerivedStateFromError(): { hasError: true } {
 		return { hasError: true };
 	}
 
-	componentDidCatch( error: Error, errorInfo: ErrorInfo ) {
+	componentDidCatch( error: Error, errorInfo: ErrorInfo ): void {
 		if ( this.props.onError ) {
 			const errorMessage = `${ error.message }; Stack: ${ error.stack }; Component Stack: ${ errorInfo.componentStack }`;
 			debug( 'reporting the error', errorMessage );
 			this.props.onError( errorMessage );
 		}
 	}
 
-	render() {
+	render(): JSX.Element | JSX.Element[] | undefined {
```

Comment:
Maybe a better return type for `render` would be `React.ReactNode` ?

(see example [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L491))